### ISSUE TITLE
How should I get superpoint when I inference on my customized dataset?

### ISSUE THREAD

**Issue by LiNY2-01** [filaPro/oneformer3d#100]
Hi! Thank you for your great work on this project. I am currently working with my own indoor point cloud data, and I would like to infer it using your pre-trained model on the ScanNet200 dataset. I am confused about how to apply superpoint clustering, as I only have points and their colors, and no mesh in my point cloud. Should I perform surface reconstruction, or should I just segment the points directly? If you have any recommendations, I would greatly appreciate it!

**Comment by filaPro** [comment]
We recommend to use ScanNet segmentator or the one from [superpoint_transformer](https://github.com/drprojects/superpoint_transformer), and as i remember both of them requires mesh as a first step.

**Comment by LiNY2-01** [comment]
Thank you for your reply! I noticed that the segmenter used in oneformer3d has segment_point method, which can be used to perform superpoint segmentation using vertices, normals and edges of the point cloud. Is this feasible? Also, do different superpoint segmentation methods affect performance?

**Comment by LiNY2-01** [comment]
You mentioned [superpoint_transformer](https://github.com/drprojects/superpoint_transformer). Do you mean the preprocessing part of superpoint_transformer? I looked at the preprocessing pipeline and found that mesh is not required, as the pipeline processes the S3DIS dataset directly without mesh construction. Could the pipeline from superpoint_transformer potentially be a solution to my issue?

**Comment by filaPro** [comment]
Yes, the preprocessing part, and you are right, it doesn't require meshes. You can also briefly check our latest [UniDet3D](https://github.com/filapro/unidet3d) where we train a single model on superpoints from ScanNet segmentor and superpoint_transformer.
